Mathlib/LinearAlgebra/InvariantBasisNumber.lean

```diff
@@ -11,34 +12,69 @@ import Mathlib.RingTheory.PrincipalIdealDomain
 /-!
 # Invariant basis number property
 
-We say that a ring `R` satisfies the invariant basis number property if there is a well-defined
-notion of the rank of a finitely generated free (left) `R`-module. Since a finitely generated free
-module with a basis consisting of `n` elements is linearly equivalent to `Fin n → R`, it is
-sufficient that `(Fin n → R) ≃ₗ[R] (Fin m → R)` implies `n = m`.
+## Main definitions
 
-It is also useful to consider two stronger conditions, namely the rank condition,
-that a surjective linear map `(Fin n → R) →ₗ[R] (Fin m → R)` implies `m ≤ n` and
-the strong rank condition, that an injective linear map `(Fin n → R) →ₗ[R] (Fin m → R)`
-implies `n ≤ m`.
+Let `R` be a (not necessary commutative) ring.
 
-The strong rank condition implies the rank condition, and the rank condition implies
-the invariant basis number property.
+- `InvariantBasisNumber R` is a type class stating that `(Fin n → R) ≃ₗ[R] (Fin m → R)`
+  implies `n = m`, a property known as the *invariant basis number property.*
 
-## Main definitions
+  This assumption implies that there is a well-defined notion of the rank
+  of a finitely generated free (left) `R`-module.
+
+It is also useful to consider the following stronger conditions:
+
+- the *rank condition*, witnessed by the type class `RankCondition R`, states that
+  the existence of a surjective linear map `(Fin n → R) →ₗ[R] (Fin m → R)` implies `m ≤ n`
+
+- the *strong rank condition*, witnessed by the type class `StrongRankCondition R`, states
+  that the existence of an injective linear map `(Fin n → R) →ₗ[R] (Fin m → R)`
+  implies `n ≤ m`.
+
+- `OrzechProperty R`, defined in `Mathlib/RingTheory/OrzechProperty.lean`,
+  is a type class stating that `R` satisfies the following property:
+  for any finitely generated `R`-module `M`, any surjective homomorphism `f : N → M`
+  from a submodule `N` of `M` to `M` is injective.
+  It was introduced in papers by Orzech [orzech1971], Djoković [djokovic1973] and
+  Ribenboim [ribenboim1971], under the names `Π`-ring or `Π₁`-ring.
+  It implies the strong rank condition if the ring is nontrivial.
+
+
+## Instances
 
-`StrongRankCondition R` is a type class stating that `R` satisfies the strong rank condition.
-`RankCondition R` is a type class stating that `R` satisfies the rank condition.
-`InvariantBasisNumber R` is a type class stating that `R` has the invariant basis number property.
+- `IsNoetherianRing.orzechProperty` (defined in `Mathlib/RingTheory/Noetherian.lean`) :
```

Comment:
I believe this part of the docstring can be removed until these properties exist.